include/wsrep/server_service.hpp

```diff
@@ -190,6 +189,22 @@ namespace wsrep
          */
         virtual void debug_sync(const char* sync_point) = 0;
 
+        /**
+         * Set encryption key
+         */
+        virtual int set_encryption_key(const unsigned char* key, 
+                                       int key_length) = 0;
+
+        /*
+         * Encrypt/decrypt data
+         */
+        virtual void do_crypt(void**                ctx,
+                              wsrep::const_buffer&  key,
+                              const char            (*iv)[32],
+                              wsrep::const_buffer&  input,
+                              void*                 output,
```

Comment:
There is no indication how big the output buffer is. It would good to add it to allow sanity checking if the caller has indeed allocated enough memory for output.

Reply:
Agree, this would be ok to do, but i think it is safe to assume that output buffer should be at least 2 * BLOCK_SIZE to avoid overflows.
I would rather prefer that database is one which creates array.

Reply:
There is no way to know output size, either interface should be changed to provide this or array should be dynamically created inside do_crypt function and released in galera.